### DISCUSSION TITLE
Using sarif in GHA?

### DISCUSSION THREAD

**Discussion by lazka** [zizmorcore/zizmor#291] [Q&A]
I' tried to set up security scanning with sarif as describe in the docs: https://github.com/msys2/msys2-autobuild/blob/1bc0a28e352f10c2da458fafa4a919c4dcf77ff3/.github/workflows/test.yml#L42-L64

I then created a test PR which introduces a warning, but it's not showing anything:  https://github.com/msys2/msys2-autobuild/pull/103 (correction: it showed the warning on the separate code scanning page for that PR, but there was no indication of any warnings on the PR page itself)

Does anyone know what I'm missing?

**Comment by woodruffw** [comment]
Hmm, that setup looks right -- could you try running without `--format=sarif` and see if the normal output produces the finding there?

Also, you do you see anything here? <https://github.com/msys2/msys2-autobuild/security/code-scanning?query=pr%3A102+is%3Aopen>

(I can't see that last link since I'm not a maintainer, but findings should appear there as well.)

**Reply by lazka** [reply]
> Hmm, that setup looks right -- could you try running without `--format=sarif` and see if the normal output produces the finding there?

Yeah, it fails here: https://github.com/msys2/msys2-autobuild/pull/104

> Also, you do you see anything here? https://github.com/msys2/msys2-autobuild/security/code-scanning?query=pr%3A102+is%3Aopen

It no longer does now that I closed the PR, but it did show the alert there before ("warning: artipacked"). But I don't want to check that page manually for every PR to see if something is wrong. There is a high chance I would miss warnings that way.

> (I can't see that last link since I'm not a maintainer, but findings should appear there as well.)

Note that I've never used the code scanning feature of GitHub, so my expectations might be off.

**Reply by woodruffw** [reply]
> It no longer does now that I closed the PR, but it did show the alert there before ("warning: artipacked"). But I don't want to check that page manually for every PR to see if something is wrong. There is a high chance I would miss warnings that way.
> 
> >

Yeah, agreed -- at least on my repos the findings _do_ show up on PRs, although GitHub seems to do some weird heuristics to determine when to show them. In particular, they might not show findings until you un-draft the PR, so you might want to try that 🙂 

To prevent yourself from missing findings, I recommend using a ruleset to force the CI to fail if `zizmor` has any unresolved findings. For example, this is what I have on `zizmor` itself: the PR isn't allowed to merge if `zizmor` produces *any* findings of *any* severity:

<img width="866" alt="Screenshot 2024-12-13 at 3 48 19 PM" src="https://github.com/user-attachments/assets/837b1634-46ad-4766-a861-e6f7f406a24f" />

**Reply by lazka** [reply]
huh, thanks, I'll try if that changes anything

**Reply by lazka** [reply]
I've created a rule targeting all branches:

![image](https://github.com/user-attachments/assets/defc05d0-bb58-48da-ad15-6f876544565b)

Then I created a new PR: https://github.com/msys2/msys2-autobuild/pull/105

There is a warning on the code scanning page for that PR (https://github.com/msys2/msys2-autobuild/security/code-scanning?query=pr%3A105+tool%3Azizmor+is%3Aopen+):

![image](https://github.com/user-attachments/assets/d2d2f239-01f5-4940-b8f1-8260ee3c0b7f)

but the PR shows nothing and lets me merge it:

![image](https://github.com/user-attachments/assets/ba7e6d63-69e4-4513-a5f7-5fdd2f483afe)

(I'll just give up on this now... Thanks for your help!)

**Reply by woodruffw** [reply]
Huh, very strange. Yeah, I don't know why that is, sorry. I'll leave this open so that someone else can hopefully chime in!

Thanks for your detailed responses!
